core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -860,6 +861,29 @@ public RexNode convertCall(
             : operandTypeChecker.getConsistency();
     final List<RexNode> exprs = convertOperands(cx, call, consistency);
     RelDataType type = rexBuilder.deriveReturnType(op, exprs);
+
+    //generate relation for `=(Row1,Row2)`
+    if (op.kind == SqlKind.EQUALS) {
+      RexNode expr0 = RexUtil.removeCast(exprs.get(0));
+      RexNode expr1 = RexUtil.removeCast(exprs.get(1));
```

Comment:
Actually why need to remove cast?

Reply:
If you have to remove cast, then better to add a test that include CAST to show the difference?

Reply:
ROW operation naturally includes a Cast outside. so the example test already has it.